### DISCUSSION TITLE
Fastapi app deployment with static files and Jinja2 Templates

### DISCUSSION THREAD

**Discussion by opticrew** [deta/cloud-docs#561] [Help] [answered]
I want to deploy a simple website using Fastapi with static files and Jinja2 Templates.

I have already successfully created a micro but with statitc files and templates serving html it's more difficult for me to understand the logic.

Can someone help me ?

The structure of my project
a folder "static"
a folder "templates"
a file main.py
a file requirements.txt

**Comment by Sokhavuth** [comment]
You must create .env file in which you must write at least:
```
BINARY_CONTENT_TYPES=image/*,font/*
```
After creating this .env file, you need to update your micro by writing:
```
deta update -e .env
```
https://docs.deta.sh/docs/micros/env_vars

**Reply by opticrew** [reply]
It works without creating .env file but with using the Jinja2 url_for tag in my html template.

I've successifully installed tailwindcss with the cli method from tailwind in my project . The css styles from tailwind work perfectly on my local computer but they dont apply on deta after deta deploy.

Have you a suggestion for me to try ??

**Reply by AyushSehrawat** [reply]
Can you send the structure and how you serve those files ( also how you output)?

**Reply by opticrew** [reply]
![Capture](https://user-images.githubusercontent.com/24733447/198816959-154f66cd-75f9-42b1-95f1-1c089456ca8c.png)

**Reply by opticrew** [reply]
it's my structure et how I serve my files

**Comment by opticrew** [comment, marked as answer]
Do you have a method to install and deploy tailwindcss successfully on deta for a fastapi app serving Jinja2 html templates ?

**Reply by AyushSehrawat** [reply]
One way is to install locally and build and put the output in a dir, then serve those files via jinja2 (You can use {{}} and other stuff for dynamic website)

Other way is using GitHub actions to automatically build... Also there exists a pypi package for tailwindcss, check it out

**Reply by opticrew** [reply]
"One way is to install locally and build and put the output in a dir, then serve those files via jinja2 (You can use {{}} and other stuff for dynamic website)"
It's what I've done and it's working perfectly locally but not after deta deployment.

**Reply by AyushSehrawat** [reply]
What's the issue? That href ain't working?

**Reply by AyushSehrawat** [reply]
If yes then die you correctly set the css dir to the tailwindcss file?

**Reply by AyushSehrawat** [reply]
> "One way is to install locally and build and put the output in a dir, then serve those files via jinja2 (You can use {{}} and other stuff for dynamic website)" It's what I've done and it's working perfectly locally but not after deta deployment.

If it works locally, can you try `deta logs` to see what's it responding on serving files?

**Reply by opticrew** [reply]
My logs indicate "no errors"...

**Reply by opticrew** [reply]
It worked, I had just to delete my navigation history in my browser ...
